services/task-svc/internal/patient/patient.go

```diff
@@ -379,3 +380,70 @@ func (ServiceServer) GetPatientDetails(ctx context.Context, req *pb.GetPatientDe
 		Tasks:                   mappedTasks,
 	}, nil
 }
+
+func (ServiceServer) GetPatientList(ctx context.Context, req *pb.GetPatientListRequest) (*pb.GetPatientListResponse, error) {
+	db := hwgorm.GetDB(ctx)
+
+	id, err := uuid.Parse(req.OrganisationId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	var unassignedPatients []patientModels.Patient
+	if err := db.Where("organization_id = ? AND bed_id IS NULL AND is_discharged = 0", id).Find(&unassignedPatients).Error; err != nil {
+		return nil, err
+	}
+
+	var dischargedPatients []patientModels.Patient
+	if err := db.Where("organization_id = ? AND is_discharged = 1", id).Find(&dischargedPatients).Error; err != nil {
+		return nil, err
+	}
+
+	var activePatients []patientModels.Patient
+	if err := db.Where("organization_id = ? AND bed_id IS NOT NULL AND is_discharged = 0", id).Find(&activePatients).Error; err != nil {
+		return nil, err
+	}
+
+	resActivePatients, err := hwutil.MapWithErr(activePatients, func(patient patientModels.Patient) (*pb.GetPatientListResponse_PatientWithRoomAndBed, error) {
+		var bed bedModels.Bed
+		var room roomModels.Room
+
+		db.Where("beds.id = ?", patient.BedID).
+			Joins("JOIN rooms ON beds.room_id = rooms.id").
+			First(&bed).
+			First(&room)
```

Comment:
its still a query for every patient, which would be very costly for a ward/organization with 50 patients or so